zio-schema-json/shared/src/main/scala/zio/schema/codec/JsonCodec.scala

```diff
@@ -222,20 +224,35 @@ object JsonCodec extends Codec {
         }
     }
 
-    private def enumEncoder[Z](cases: Schema.Case[Z, _]*): JsonEncoder[Z] =
+    private def enumEncoder[Z](parentSchema: Schema.Enum[Z], cases: Schema.Case[Z, _]*): JsonEncoder[Z] =
       (value: Z, indent: Option[Int], out: Write) => {
         val fieldIndex = cases.indexWhere(c => c.deconstructOption(value).isDefined)
         if (fieldIndex > -1) {
           val case_ = cases(fieldIndex)
-          out.write('{')
+          val discriminatorChunk = parentSchema.annotations.collect {
+            case d: discriminatorName =>
+              parentSchema.id match {
+                case TypeId.Structural => None
+                case n: TypeId.Nominal => Some((d, (n.packageName ++ n.objectNames ++ Chunk(case_.id)).mkString(".")))
```

Comment:
Rather than using a type ID, we should use the `id` of the `Case` of the enumeration that we are encoding. That is used even when discriminators are not used to identify which of the case in the enum a value is.